File: scala/word-count/src/main/scala/wordcount.scala
class Phrase(phrase: String) {
  def wordCount(): Map[String, Int] = {

   "[\\w']+".r
     .findAllIn(phrase)
     .foldLeft(Map[String, Int]())({(map, word) => map.get(word) match {
        case None => map + ((word, 1))
        case Some(i) => map.updated(word, i +1)
      }
     })

  }
}

File: python/word-count/word_count.py
import re


def word_count(phrase):
    words = filter(None, re.sub('[^a-z0-9 ]', ' ', phrase.lower()).split(' '))
    dct = {}
    for word in words:
        if word in dct:
            dct[word] += 1
        else:
            dct[word] = 1
    return dct
